include/zfs_fletcher.h

```diff
@@ -45,6 +45,21 @@ void fletcher_4_incremental_native(const void *, uint64_t,
     zio_cksum_t *);
 void fletcher_4_incremental_byteswap(const void *, uint64_t,
     zio_cksum_t *);
+int fletcher_4_impl_set(const char *selector);
+void fletcher_4_init(void);
+void fletcher_4_fini(void);
+
+/*
+ * fletcher checksum struct
+ */
+struct fletcher_4_calls {
+	void (*init)(zio_cksum_t *);
+	void (*fini)(zio_cksum_t *);
+	void (*compute)(const void *, uint64_t, zio_cksum_t *);
+	void (*compute_byteswap)(const void *, uint64_t, zio_cksum_t *);
+	boolean_t (*valid)(void);
+	const char *name;
+};
```

Comment:
There should be a typedef for this which is defined and then used throughout the code.

```
typedef struct fletcher_4_calls {
	void (*init)(zio_cksum_t *);
	void (*fini)(zio_cksum_t *);
	void (*compute)(const void *, uint64_t, zio_cksum_t *);
	void (*compute_byteswap)(const void *, uint64_t, zio_cksum_t *);
	boolean_t (*valid)(void);
	const char *name;
} fletcher_4_calls_t;
```

[style] naming this `fletcher_4_func_t` would be consistent with the rest of the code base.  The `_calls` suffix isn't used anywhere to imply a set of callbacks.